File: BroadcastReciever/app/src/main/java/com/example/broadcastreciever/MainActivity.java
package com.example.broadcastreciever;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
  TextView textView;
  MyBroadcastReciever myBroadcastReciever;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView=findViewById(R.id.tv);
        myBroadcastReciever=new MyBroadcastReciever(textView);
        registerReceiver(myBroadcastReciever,new IntentFilter(Intent.ACTION_BATTERY_CHANGED));

    }

    @Override
    protected void onStop() {
        super.onStop();
        unregisterReceiver(myBroadcastReciever);
    }
}

File: MyNewsApi/app/src/main/java/com/example/mynewsapi/MainActivity.kt
package com.example.mynewsapi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val songsObjects:MutableList<Song> = mutableListOf<Song>()
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        songsObjects.add(Song("Akram","rrrrrrrrrrrrrr"))
        val recyclerView :RecyclerView
        recyclerView=findViewById(R.id.recycler)
        recyclerView.adapter=MyAdapter(songsObjects)
        recyclerView.layoutManager=LinearLayoutManager(this)
    }
}